sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -599,6 +599,38 @@ class Dataset[T] private[sql](
    */
   def takeAsList(num: Int): java.util.List[T] = java.util.Arrays.asList(take(num) : _*)
 
+  /**
+   * Compose the string representing rows for output
+   * @param _numRows Number of rows to show
+   * @param truncate Whether truncate long strings and align cells right
+   */
+  override private[sql] def showString(_numRows: Int, truncate: Boolean = true): String = {
```

Comment:
When merging them into the same one, I hit a problem when calling the function `take`. The `Dataset` API has different type `Array[T]`, but `Dataframe` API has a type `Array[T]`. 

Do you have a better idea how to integrate into the same function?